### DISCUSSION TITLE
ARM docker / Plain Opensearch

### DISCUSSION THREAD

**Discussion by ghost** [wazuh/wazuh#19862] [Q&A]
Hello all,

I'd like to deploy Wazuh on an arm64 / ampere server as a container deployment. Unfortunately there are no images for the indexer. As far as I know the indexer is Opensearch and the Web UI is Opensearch Dashboards with a plugin.
Therefore I'd like to know if it is possible to use plain Opensearch as the indexer and Opensearch Dashboards with the plugin installed?

Thank you in advance for your assistance

**Comment by mdiego92** [comment]
Hello @iFrozenPhoenix,

You can build your own images locally for that. Please find the required documentation below:
https://documentation.wazuh.com/current/deployment-options/docker/wazuh-container.html#build-docker-images-locally

Let me know if that's what you need!
Looking forward to your feedback.

Regards.

**Reply by ghost** [reply]
@mdiego92 thank you for your answer. Unfortunately the build fails to download the indexer packages. According to the docs https://documentation.wazuh.com/current/installation-guide/packages-list.html there are no arm64 packages for the indexer.

**Reply by mdiego92** [reply]
Hello @iFrozenPhoenix,

It is not possible to run natively the Wazuh Dashboard on arm64.
It's currently being worked on.

A possible workaround could be running a Wazuh Manager installation, and then manually installing OpenSearch and OpenSearch Dashboards to replace Wazuh Indexer and Dashboard, respectively.
Reference: https://documentation.wazuh.com/current/integrations-guide/opensearch/index.html

The Wazuh App plugin is not available for arm64, so you won't be able to manage agents from the UI, and you will be limited from visualizing alert data.

Another issue you may have with this is running it through Docker, as the Filebeat component used by Wazuh is not compatible with ARM architectures.

I'm sorry for the trouble.
Please let me know if there is anything else I can do from my end.

Regards.

**Reply by ghost** [reply]
Thank you for your answer @mdiego92,

I want to try get it running. I would appreciate if you could tell me why it is not possible to run the Wazuh Dashboard on ARM? As far as i know the Wazuh Dashboard is OpenSearch Dashboards with the wazuh-dashboards-plugin installed. The Source code of the plugin looks like it has only portable TS/JS code (I may be wrong). Shouldn't it work if i only install the wazuh-dashboards-plugin into an ARM OpenSeach Dashboards installation (Which is available and working)?
Concerning filebeat i will try if it is possible to use the latest version. If this is not working I will compile the version that you are using myself.
So the process would be: Using plain OpenSearch and OpenSearch Dashboards, install the wazuh-dashboards-plugin, install filebeat on indexer. Is this correct?
